src/harness/harness.ts

```diff
@@ -1184,56 +1229,39 @@ namespace Harness {
-                const builtFileName = ts.combinePaths(libFolder, options.includeBuiltFile);
-                const builtFile: TestFile = {
-                    unitName: builtFileName,
-                    content: normalizeLineEndings(IO.readFile(builtFileName), Harness.IO.newLine()),
-                };
-                programFiles.push(builtFile);
+                programFileNames.push(vpath.combine("/.ts/", options.includeBuiltFile));
             }
 
-            const fileOutputs: GeneratedFile[] = [];
-
             // Files from tests\lib that are requested by "@libFiles"
             if (options.libFiles) {
                 for (const fileName of options.libFiles.split(",")) {
-                    const libFileName = "tests/lib/" + fileName;
-                    // Content is undefined here because in createCompilerHost we will create sourceFile for the lib file and cache the result
-                    programFiles.push({ unitName: libFileName, content: undefined });
-                }
-            }
-
-
-            const programFileNames = programFiles.map(file => file.unitName);
-
-            const compilerHost = createCompilerHost(
-                programFiles.concat(otherFiles),
-                (fileName, code, writeByteOrderMark) => fileOutputs.push({ fileName, code, writeByteOrderMark }),
-                options.target,
-                useCaseSensitiveFileNames,
-                currentDirectory,
-                options.newLine,
-                options.libFiles);
-
-            let traceResults: string[];
-            if (options.traceResolution) {
-                traceResults = [];
-                compilerHost.trace = text => traceResults.push(text);
-            }
-            else {
-                compilerHost.directoryExists = () => true; // This only visibly affects resolution traces, so to save time we always return true where possible
-            }
-            const program = ts.createProgram(programFileNames, options, compilerHost);
-
-            const emitResult = program.emit();
-
-            const errors = ts.getPreEmitDiagnostics(program);
-
-            const result = new CompilerResult(fileOutputs, errors, program, Harness.IO.getCurrentDirectory(), emitResult.sourceMaps, traceResults);
+                    programFileNames.push(vpath.combine("/.lib/", fileName));
+                }
+            }
+
+            const result = compiler.compileFiles(
+                new compiler.CompilerHost(
+                    vfs.VirtualFileSystem.createFromDocuments(
+                        useCaseSensitiveFileNames,
+                        inputFiles.concat(otherFiles).map(documents.TextDocument.fromTestFile),
+                        { currentDirectory, overwrite: true }
+                    ),
+                    options
+                ),
+                programFileNames,
+                options);
+
+            // const fileOutputs = compilation.outputs.map(output => output.asGeneratedFile());

```

Comment:
Is the commented-out code here required in some way?

Reply:
Not anymore, I'll remove it.